test/test_cgp_evolve.py

```diff
@@ -217,25 +217,34 @@ def test_speedup_parallel_evolve():
+    n_calls_objective = (population_params['n_parents'] +
+                         (population_params['n_parents'] +
+                          population_params['n_offsprings']) *
+                         (population_params['max_generations'] - 1))
     np.random.seed(SEED)
 
-    pop_one_thread = gp.CGPPopulation(population_params['n_parents'], population_params['mutation_rate'], SEED, genome_params)
-    ea = gp.ea.MuPlusLambda(population_params['n_offsprings'], population_params['n_breeding'], population_params['tournament_size'])
-
-    t0 = time.time()
-    gp.evolve(pop_one_thread, objective_speedup_parallel_evolve, ea, population_params['max_generations'], population_params['min_fitness'])
-    assert time.time() - t0 == pytest.approx((population_params['n_parents'] + population_params['n_offsprings']) * population_params['max_generations'] * 0.1, rel=0.25)
-
-    pop_two_threads = gp.CGPPopulation(population_params['n_parents'], population_params['mutation_rate'], SEED, genome_params)
-    ea = gp.ea.MuPlusLambda(population_params['n_offsprings'], population_params['n_breeding'], population_params['tournament_size'], n_processes=2)
-
-    t0 = time.time()
-    gp.evolve(pop_two_threads, objective_speedup_parallel_evolve, ea, population_params['max_generations'], population_params['min_fitness'])
-    assert time.time() - t0 == pytest.approx((population_params['n_parents'] + population_params['n_offsprings']) * population_params['max_generations'] * 0.1 / 2., rel=0.25)
-
-    pop_4_threads = gp.CGPPopulation(population_params['n_parents'], population_params['mutation_rate'], SEED, genome_params)
-    ea = gp.ea.MuPlusLambda(population_params['n_offsprings'], population_params['n_breeding'], population_params['tournament_size'], n_processes=4)
-
-    t0 = time.time()
-    gp.evolve(pop_4_threads, objective_speedup_parallel_evolve, ea, population_params['max_generations'], population_params['min_fitness'])
-    assert time.time() - t0 == pytest.approx((population_params['n_parents'] + population_params['n_offsprings']) * population_params['max_generations'] * 0.1 / 4., rel=0.25)
+    # Serial execution
+    
+    for n_processes in [1, 2, 4]:
+        pop = gp.CGPPopulation(population_params['n_parents'],
+                               population_params['mutation_rate'], SEED, genome_params)
+        ea = gp.ea.MuPlusLambda(population_params['n_offsprings'],
+                                population_params['n_breeding'],
+                                population_params['tournament_size'],
+                                n_processes=n_processes)
+
+        t0 = time.time()
+        gp.evolve(pop, objective_speedup_parallel_evolve, ea,
+                  population_params['max_generations'],
+                  population_params['min_fitness'])
+        T = time.time() - t0
+        print(T)
+        if n_processes == 1:
+            T_baseline = T
+            assert T == pytest.approx(n_calls_objective * 0.1, rel=0.25)
```

Comment:
i don't understand this assert, maybe add a comment?

Reply:
This assert tests whether the total time is roughly equal to number of calls to the objective x execution time of one objective call.

Same as before, just refactored to make it more readable.

Reply:
oh wow, the `0.1` comes from the `sleep(0.1)`, i didn't see that, ok